wp-includes/service-workers.php

```diff
@@ -450,3 +451,10 @@ function wp_service_worker_get_error_messages() {
 function wp_service_worker_error_message_placeholder() {
 	echo '<p><!--WP_SERVICE_WORKER_ERROR_MESSAGE--></p>'; // WPCS: XSS OK.
 }
+
+/**
+ * Display service worker error details template tag.
+ */
+function wp_service_worker_error_details_placeholder() {
```

Comment:
@miina Here's an idea that could tie things together. What if this template tag tags arguments to construct the `details` element (or another element) to populate? This function could then also add a comment before and after the output element which could then facilitate removing the `details` entirely if there is no error message to display.

Consider this usage:

```php
wp_service_worker_error_details_template(
    // The following could be the default value.
    '<details id="error-details"><summary>' . esc_html__( 'More Details', 'pwa' ) . '</summary>{{{error_details_iframe}}}</details>' 
);
```

This could then output markup like this:

```html
<!--WP_SERVICE_WORKER_ERROR_TEMPLATE_BEGIN-->
<details id="error-details"><summary>More details</summary>{{{error_details_iframe}}}</details>
<!--WP_SERVICE_WORKER_ERROR_TEMPLATE_END-->
```

Then the service worker can do a regex replacement:

```js
text.replace(
    /(<!--WP_SERVICE_WORKER_ERROR_TEMPLATE_BEGIN-->)((?:.|\n)+?)(<!--WP_SERVICE_WORKER_ERROR_TEMPLATE_END-->)/, 
    ( matches ) => {
        if ( ! errorText ) {
            return ''; // Remove the details from the document entirely.
        }
        let details = matches[1];
        const src = '...'; // The errorText encoded as a text/html data URL.
        const srcdoc = '...'; // The errorText escaped for use in an HTML attribute.
        const iframe = `<iframe style="width:100%" src="${src}"  srcdoc="${srcdoc}"></iframe>`;
        details = details.replace( '{{{error_details_iframe}}}', iframe );
        // The following are in case the user wants to include the <iframe> in the template.
        details = details.replace( '{{{iframe_src}}}', src );
        details = details.replace( '{{{iframe_srcdoc}}}', srcdoc );
        return details;
    } 
);
```

Reply:
Made some changes within d89b536. Let me know if the data URL / escaped attributes seem to be correct.

Reply:
One key thing when doing HTML escaping is that the `&` has to be always replaced with `&amp;` first. Otherwise, you could end up with `<` escaped to `&lt;` and then double-escaped to`&amp;lt;`.